### PR TITLE
External consumers implementation

### PR DESCRIPTION
**Pull request by riccardomodanese**
Brief description of the PR.
This pr implements the architectural change moving the business logic from the broker to a couple of consumers based on Spring Boot/Camel

**Related Issue**
none

**Description of the solution adopted**
Before this pr the business logic was implemented through Camel routes running inside broker jvm.
Now the telelemetry messages processing logic is moved to a "telemetry" consumer and the lifecycle messages processing logic to a "lifecycle" consumer.
Both consumers are based on Spring Boot, expose health check and metrics through a couple of registered servlet and start the proper Camel routes inside the Camel context.
The standard deployment provide one consumer of both types associated to the broker.

**Screenshots**
none

**Any side note on the changes made**
I'm asking for a deep code review and also a manual test of all the features by reviewers (especially those less covered by unit/integration tests).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,7 +52,7 @@ The team maintains some docker images in a Docker Hub repository at [Kapua Repos
 **Note:** the Docker Hub repository mentioned above is not the official project repository from Eclipse Foundation.
 ***
 
-Suppose the target is the current snapshot 1.6.0-SNAPSHOT.
+Suppose the target is the current snapshot 1.6.0-CAMEL-SNAPSHOT.
```
Comment (lorthirk) [inline README.md:55]: Correct the Kapua version

File: transport/pom.xml [hunk 2]
```diff
@@ -18,7 +18,7 @@
     <parent>
         <groupId>org.eclipse.kapua</groupId>
         <artifactId>kapua</artifactId>
-        <version>1.6.0-SNAPSHOT</version>
+        <version>1.6.0-CAMEL-SNAPSHOT</version>
```
Comment (Coduz) [inline transport/pom.xml]: Rename as 1.6.0-SNAPSHOT

File: tmpCode/ErrorMessageListener.j [hunk 3]
```diff
@@ -0,0 +1,167 @@
+/*******************************************************************************
```
Comment (Coduz) [inline tmpCode/ErrorMessageListener.j]: Is this to be removed?

File: tmpCode/DataStorageMessageProcessor.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
```
Comment (Coduz) [inline tmpCode/DataStorageMessageProcessor.java]: Is this to be removed?

File: service/security/shiro/src/test/java/org/eclipse/kapua/service/authentication/shiro/mfa/MfaAuthenticatorImplTest.java [hunk 5]
```diff
@@ -41,67 +41,67 @@ public void initialize() throws KapuaException {
...
-        }
-    }
-
-    @Test
-    public void authorizeHasedScratchCodeVerificationNullCodeParametersTest() {
-        for (String hasedScratchCode : hashedScratchCodes) {
-            assertFalse("False expected.", mfaAuthenticatorImpl.authorize(hasedScratchCode, null));
-        }
-    }
+//    @Test(expected = NullPointerException.class)
```
Comment (Coduz) [inline service/security/shiro/src/test/java/org/eclipse/kapua/service/authentication/shiro/mfa/MfaAuthenticatorImplTest.java]: Please remove commented tests

File: service/security/shiro/src/test/java/org/eclipse/kapua/service/authentication/shiro/mfa/MfaAuthenticatorImplTest.java [hunk 6]
```diff
@@ -41,67 +41,67 @@ public void initialize() throws KapuaException {
...
         }
     }
 
-    @Test(expected = IllegalArgumentException.class)
-    public void authorizeEncryptedSecretNullVerificationCodeParametersTest() {
-        for (String encryptedSecret : encryptedSecrets) {
-            assertFalse("False expected.", mfaAuthenticatorImpl.authorize(encryptedSecret, null));
-        }
-    }
+//    @Test(expected = IllegalArgumentException.class)
```
Comment (Coduz) [inline service/security/shiro/src/test/java/org/eclipse/kapua/service/authentication/shiro/mfa/MfaAuthenticatorImplTest.java]: Please remove commented tests

File: assembly/jetty-base/pom.xml [hunk 7]
```diff
@@ -19,14 +19,14 @@
...
-        <version>1.6.0-SNAPSHOT</version>
+        <version>1.6.0-CAMEL-SNAPSHOT</version>
     </parent>
 
     <packaging>pom</packaging>
     <artifactId>kapua-assembly-jetty-base</artifactId>
 
     <properties>
-        <jetty.url>https://repo1.maven.org/maven2/org/eclipse/jetty/jetty-distribution/9.4.12.v20180830/jetty-distribution-9.4.12.v20180830.tar.gz</jetty.url>
+        <jetty.url>https://repo1.maven.org/maven2/org/eclipse/jetty/jetty-distribution/9.4.39.v20210325/jetty-distribution-9.4.39.v20210325.tar.gz</jetty.url>
```
Comment (Coduz) [inline assembly/jetty-base/pom.xml]: Would maybe better to replace with ${jetty.version} ?

**Review by Coduz — CHANGES_REQUESTED** [review]

**Review by Coduz — APPROVED** [review]

**Pull request merged**